controllers/Challenges.js

```diff
@@ -2,42 +2,47 @@ const email = require('../utils').email;
 const { logger } = require('../utils');
 const models = require('../models');
 const Challenge = new models.Challenge();
-const Performance = new models.Performance();
+const Performance = models.Performance;
 
 function ChallengersController() {
   this.create = (req, res) => {
     const { spotId } = req.body;
-    const userId = req.user.nameNumber, performanceId = req.session.currentPerformance.id;
+    const userId = req.user.nameNumber;
 
-    logger.challengesLog(`${req.user.name} sent request to challenge ${spotId}`);
-    return Challenge.create(userId, spotId, performanceId)
-      .then((code) => {
-        email.sendChallengeSuccessEmail({
-          email: req.user.email,
-          performanceName: req.session.currentPerformance.name,
-          spotId
-        });
-        logger.challengesLog(`${req.user.name} successfully challenged for ${spotId}`);
-        res.json({ code });
-      })
-      .catch((err) => {
-        logger.errorLog('Challenges.create', err);
-        res.status(400).send(err);
-      });
+    return Performance.findNextOrOpenWindow()
```

Comment:
There's really no performance bump, correct? I'd rather have it explicitly called by the controllers that need it rather than have it be partially hidden in middleware. If I find that I'm using it more and more (now that the model class is much better 🎉 ) I'll switch to middleware